src/ray/raylet/lineage_cache.cc

```diff
@@ -256,15 +264,21 @@ void LineageCache::RemoveWaitingTask(const TaskID &task_id) {
   }
 }
 
-Lineage LineageCache::GetUncommittedLineage(const TaskID &task_id) const {
+void LineageCache::MarkTaskAsForwarded(const TaskID &task_id, const ClientID &node_id) {
+  lineage_.GetEntryMutable(task_id)->MarkExplicitlyForwarded(node_id);
```

Comment:
Since we're using `ClientID::nil()` to mean "get all the uncommitted lineage", we should probably add a `RAY_CHECK` here to make sure that `!node_id.is_nil()`.